tests/server.py

```diff
@@ -134,8 +134,12 @@ def process(self) -> None:
                 file_content = None
                 try:
                     file_content = (static_path / path[1:]).read_bytes()
-                    request.setHeader(b"Content-Type", mimetypes.guess_type(path)[0])
+                    content_type = mimetypes.guess_type(path)[0]
+                    if content_type and content_type.startswith("text/"):
+                        content_type += "; charset=utf-8"
+                    request.setHeader(b"Content-Type", content_type)
                     request.setHeader(b"Cache-Control", "no-cache, no-store")
+                    request.setHeader(b"Content-Length", str(len(file_content)))
```

Comment:
These changes were made to match upstream Node.js server. Without them, FF has different behavior than the other browsers.